### DISCUSSION TITLE
5.1.0 Planning

### DISCUSSION THREAD

**Discussion by jonmmease** [vega/altair#3145] [General]
I wanted to start a discussion to plan what remaining work we would like to accomplish before releasing version 5.1.0.

I would like to propose that we release 5.1 before the Vega conference (September 18th) at the latest, but I don't think we necessarily need to wait that long.

For myself, I have a couple of documentation PRs open that I'd like to get in. And I want to release VegaFusion 1.4.0 before Altair 5.1.0 and then update Altair's optional dependency on VegaFusion to be at least version 1.4.0 (this way I don't need to worry about difference in VegaFusion versions right away).

What else?

@binste, do we want to aim to finish the typing effort and make 5.1.0 fully typed?

**Comment by mattijn** [comment]
Update to a later Vega-Lite version. Altair 5.0.1 is on VL 5.8.0, but the latest VL version is 5.14.1 currently. I assume our compagnons at the VL repo will try to make a new release too.

This requires currently, also changes to the `altair_viewer` master branch to make the CI tests pass. Maybe we could solve that too. For me it is a bit strange that we have to update this downstream package to make Altair tests works.

**Reply by jonmmease** [reply]
Ahh, yeah, updating Vega-Lite is a good idea.

I don't recall what the `altair_viewer` dependency is. Is that for offline html generation?

**Reply by binste** [reply]
Exactly, offline html generation, offline chart viewing, and also rendering of charts in a separate browser window which allows users to work in a terminal environment such as `ipython`. Unfortunately, we don't have publishing rights on pypi for it.

I'll create a PR to remove this dependency for testing.

**Reply by binste** [reply]
See in #3147 re altair_viewer. I'll work on another PR now to update VL to 5.14.1.

**Comment by binste** [comment]
This plan sounds good to me! I'd need a review on https://github.com/altair-viz/altair/pull/3118 which I think should be part of 5.1 so users no longer see the deprecation warning from jsonschema. Regarding the type hints, I think this requires more time then I could put in in the next 3-4 weeks. Especially #3143 and #2854 still need quite some work, probably more something for a 5.2 release. Does that work for you @jonmmease or do you have a specific use case where you or anyone would benefit from 5.1 already being fully typed?

I'm ok with disabling the tests which rely on `altair_viewer` as we can't release a version for Altair 5 anyway so no need to test it.

**Reply by jonmmease** [reply]
> I think this requires more time then I could put in in the next 3-4 weeks. Especially https://github.com/altair-viz/altair/pull/3143 and https://github.com/altair-viz/altair/issues/2854 still need quite some work, probably more something for a 5.2 release.

Sounds good! I'm not in a rush, just wanted to ask if it was something close enough that you'd want to try to get it in for 5.1.

**Comment by jonmmease** [comment]
Ok, I think everything in this thread has been merged other than my VegaFusion 1.4.0 release and minimum version update. I'll plan to do both of these this coming week, and then perhaps we could do an Altair release the following week (week of the 20th).  @mattijn would that work for your schedule (assuming you will be the one to cut the release).

Also, do we want to do a release candidate this time around?

**Reply by mattijn** [reply]
Sounds good! I can cut a release in week 34👍

**Reply by mattijn** [reply]
I don't think a release candidate is necessary.

**Reply by jonmmease** [reply]
I released VegaFusion earlier today, and the last changes I'd like included in 5.1 are in https://github.com/altair-viz/altair/pull/3160. So I think we're on track for a release later this week.

**Comment by joelostblom** [comment]
This plans sounds good to me too. Sorry for the late chime in, between work and vacation I haven't had much spare time lately but I'm catching up on all things Altair now.

**Comment by mattijn** [comment]
I was checking the docs of the main repo a bit and I noticed that the new `JupyterChart` is still missing in the release notes around https://github.com/altair-viz/altair/blob/main/doc/releases/changes.rst?plain=1#L6.

Btw, @jonmmease, is this docs section on interactivity using VegaFusion still necessary now we have the JupyterChart in? https://github.com/altair-viz/altair/blob/main/doc/user_guide/large_datasets.rst?plain=1#L170.

**Reply by jonmmease** [reply]
Ahh, yeah I'll add JupyterChart to the release notes. Thanks for calling that out.

And yes, this VegaFusion section is still relevant since VegaFusion + JupyterChart doesn't cause selections to be evaluated on the server yet. That's a goal of mine, but we're not there yet.

**Reply by mattijn** [reply]
Ah yes! Nice. Step by step👍

**Comment by jonmmease** [comment]
I'm going to try to sneak in one more thing by EOD. I'm working on a release of vl-convert (0.13.0) that will include support for setting the PPI  (pixels per inch) of PNG images, and I'd like to update Altair to include this as the minimum vl-convert version and add the ppi argument to `chart.save(ppi=200)` and `alt.renderers.enable("png", ppi=200)`.  I have a local prototype of the altair change and it's a pretty small diff. cc @joelostblom
